modules/angular2/src/facade/collection.ts

```
@@ -10,6 +10,27 @@ export var Map = global.Map;
 export var Set = global.Set;
 export var StringMap = global.Object;
 
+// Safari and Internet Explorer do not support the iterable parameter to the
+// Map constructor.  We work around that by manually adding the items.
+var createMapFromPairs: {(pairs: List<any>): Map<any, any>} = (function() {
+  try {
+    if (new Map([1, 2]).size === 2) {
+      return function createMapFromPairs(pairs: List<any>): Map<any, any> {
+        return new Map(pairs);
+      };
+    }
+  } catch (e) {
+  }
+  return function createMapAndPopulateFromPairs(pairs: List<any>): Map<any, any> {
```

I like it outside the catch.  For style check reasons, the empty catch is required to occupy two lines :)

While the browser I tested throw an exception when passed the parameter, it's possible that some polyfill or a different browser just ignores the parameter.  That's why I have the size check.  Failing the size check also requires using the slower approach just like for the catch clause.  In the interest of DRY, I have a single return as a catch all right at the end to accomplish this.